app/controllers/actions/AuthenticatedControllerComponents.scala

```diff
@@ -42,10 +43,13 @@ trait AuthenticatedControllerComponents extends MessagesControllerComponents {
     auth andThen getRegistration
 
   def authAndGetOptionalData(period: Period): ActionBuilder[OptionalDataRequest, AnyContent] =
-    auth andThen getRegistration andThen getData(period)
+    auth andThen getRegistration andThen getData(period) andThen checkReturn
 
   def authAndGetData(period: Period): ActionBuilder[DataRequest, AnyContent] =
     authAndGetOptionalData(period) andThen requireData
+
+  def authAndGetOptionalDataNoReturnCheck(period: Period): ActionBuilder[OptionalDataRequest, AnyContent] =
```

Comment:
I don't think we actually need this one. I might be wrong but I don't believe any of the previous return pages actually require anything from userAnswers. Therefore don't need 'getData' we could just use authAndGetRegistration and delete this method? I'm just looking the previous pages now...

Reply:
Have removed redundant action.